src/core/stdc/complex.d

```diff
@@ -149,14 +149,14 @@ creal   csqrtl(creal z);
  ///
  real   cargl(creal z);
 
- ///
- double cimag(cdouble z);
- ///
- float  cimagf(cfloat z);
- ///
- real   cimagl(creal z);
+///
+pragma(inline, true) double cimag(cdouble z) { return (cast(double*)&z)[1]; }
```

Comment:
Why not `return z.im;`?

Reply:
Because `z.im` would be an `idouble`, and casting that to `double` would result in `0`.

Reply:
?? typeof(z.im) == double!